tlsn/tlsn-notary-client/src/lib.rs

```diff
@@ -0,0 +1,14 @@
+//! Notary client library.
+//!
+//! This library contains TLSNotary notary client implementations, which helps to setup
+//! connection to TLSN notary server via TCP or TLS, and subsequent requests for notarization.
```

Comment:
I thought it beneficial to be more verbose in this entry point to spell it out for the reader that notary client is not a prover per se.
(I couldn't find an elegant way to work into this sentence the idea that the notary is a "TLSNotary protocol's notary", so I omitted that part.)

lmk if this adds clarity or confusion.

```suggestion
//! A notary client's purpose is to establish a connection to the notary server via TCP or TLS, and
//! to request authorization to perform notarization. 
//! Note that the actual notarization is not performed by the notary client but by the prover of the
//! TLSNotary protocol.
```

Reply:
Thanks for the rewording! It's definitely clearer, the only thing is that I find `authorization` a little confusing as notary server has an optional authorization feature that uses API key for whitelisting — I have changed your wording slightly in my latest commit, please let me know if that's alright :)